### ISSUE TITLE
Principal in static module variables - non-static expression in library or moduleMotoko(M0014)

### ISSUE THREAD

**Issue by infu** [dfinity/motoko#4754]
Minor hindrance worth mentioning. Not something that we can't get around.

```
module {
public let default_id = Principal.fromText("lxzze-o7777-77777-aaaaa-cai");
```

results in
non-static expression in library or moduleMotoko(M0014)

There is no way to put a principal inside a module's static variables

Perhaps something like this would work, similar to how Blob works
```
module {
public let default_id = "lxzze-o7777-77777-aaaaa-cai":Principal
```

**Comment by crusso** [comment]
Yeah, that's annoying. You could declare a constant function instead.

**Comment by ggreif** [comment]
What about `let default_actor : actor {} = actor "lxzze-o7777-77777-aaaaa-cai";` and then turn that into a `Principal` at the usage site?

**Comment by infu** [comment]
nice, both options work better than what I tried - putting opt and later trapping if it's not there

**Comment by ggreif** [comment]
@crusso maybe we can mark certain functions/prims as _static-preserving_, and as such extend the set of allowed values in modules?
